### PR TITLE
Clarify 'implements ChunkStore' doc comments

### LINKED ISSUE #3085: Comments saying 'implements ChunkStore' are confusing
This for instance:

https://github.com/cortexproject/cortex/blob/ae806700415285df8c4a9e35208e25ba81235a70/pkg/chunk/series_store.go#L421

The easiest change would be to make it say `Store`.

### PR DESCRIPTION
**Pull request by bvwells**
**What this PR does**:

Clarify 'implements ChunkStore' doc comments

**Which issue(s) this PR fixes**:
Fixes #3085 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +72,7 @@
 * [BUGFIX] Ruler Experimental API: Don't allow rule groups without names or empty rule groups. #3210
 * [BUGFIX] Experimental Alertmanager API: Do not allow empty Alertmanager configurations or bad template filenames to be submitted through the configuration API. #3185
 * [BUGFIX] When using ruler sharding, moving all user rule groups from ruler to a different one and then back could end up with some user groups not being evaluated at all. #3235
+* [BUGFIX] Clarify 'implements ChunkStore' doc comments. #3085
```
Comment (bboreham) [inline CHANGELOG.md:75]: Doesn't belong in release notes as it has no impact on users.

File: pkg/querier/lazyquery/lazyquery.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func (l LazyQuerier) Close() error {
 	return l.next.Close()
 }
 
-// Get implements ChunkStore for the chunk tar HTTP handler.
+// Get implements Store for the chunk tar HTTP handler.
```
Comment (bboreham) [inline pkg/querier/lazyquery/lazyquery.go:75]: perhaps `chunk.Store` ?

File: CHANGELOG.md (hunk 1 above)
Comment (bvwells) [reply CHANGELOG.md]: Fair enough! Gone.

File: pkg/querier/lazyquery/lazyquery.go (hunk 2 above)
Comment (bvwells) [reply pkg/querier/lazyquery/lazyquery.go]: Updated.

**Comment by bvwells** [comment]
Thanks for the review @bboreham!

**Review by pstibrany — APPROVED** [review]
Thanks!

**Review by bboreham — APPROVED** [review]
Thanks!

**Review by pracucci — APPROVED** [review]
Thanks!

**Pull request merged**